### ISSUE TITLE
Trying to use on MacBook Pro M1 system: crashes on installation of the cli

### ISSUE THREAD

**Issue by raguay** [tauri-apps/tauri#2143]
### **Describe the bug**
I'm trying to install tauri in a project using:

```
npm install -D @tauri-apps/cli
```

### **To Reproduce**
1. Create a project directory. In my case, it is a Svelte project
2. In the directory, run the command:  npm install -D @tauri-apps/cli
3. After a while it crashes out.

### **Expected behavior**

The Tauri installation should finish normally and I can work on creating my application.

### **Platform and Versions (required):**

```
Operating System
macOS Big Sur on MacBook Pro M1

Node.js environment: v 16.4.1
Node.js 
@tauri-apps/cli 
@tauri-apps/api 

Global packages
npm 7.18.1 

Rust environment
rustc 1.53.0
cargo 1.53.0

App directory structure

Can't get this far.
```

### **Stack Trace**
npm install -D @tauri-apps/cli                                   09:50:02
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm ERR! code 1
npm ERR! path /Users/raguay/Dropbox (Personal)/Richard/rag/JavaScript/svelte/EmailIt/node_modules/sharp
npm ERR! command failed
npm ERR! command sh -c (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
npm ERR! sharp: Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.10.6/libvips-8.10.6-darwin-arm64v8.tar.br
npm ERR!   CC(target) Release/obj.target/nothing/../node-addon-api/nothing.o
npm ERR!   LIBTOOL-STATIC Release/nothing.a
npm ERR!   TOUCH Release/obj.target/libvips-cpp.stamp
npm ERR!   CXX(target) Release/obj.target/sharp/src/common.o
npm ERR! prebuild-install WARN install No prebuilt binaries found (target=3 runtime=napi arch=arm64 libc= platform=darwin)
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.2
npm ERR! gyp info using node@16.4.1 | darwin | arm64
npm ERR! gyp info find Python using Python version 3.9.5 found at "/opt/homebrew/opt/python@3.9/bin/python3.9"
npm ERR! gyp http GET https://nodejs.org/download/release/v16.4.1/node-v16.4.1-headers.tar.gz
npm ERR! gyp http 200 https://nodejs.org/download/release/v16.4.1/node-v16.4.1-headers.tar.gz
npm ERR! gyp http GET https://nodejs.org/download/release/v16.4.1/SHASUMS256.txt
npm ERR! gyp http 200 https://nodejs.org/download/release/v16.4.1/SHASUMS256.txt
npm ERR! (node:43109) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
npm ERR! (Use `node --trace-deprecation ...` to show where the warning was created)
npm ERR! gyp info spawn /opt/homebrew/opt/python@3.9/bin/python3.9
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/raguay/Dropbox (Personal)/Richard/rag/JavaScript/svelte/EmailIt/node_modules/sharp/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/raguay/Library/Caches/node-gyp/16.4.1/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/Users/raguay/Library/Caches/node-gyp/16.4.1',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/Users/raguay/Library/Caches/node-gyp/16.4.1/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/Users/raguay/Dropbox (Personal)/Richard/rag/JavaScript/svelte/EmailIt/node_modules/sharp',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! warning: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: archive library: Release/nothing.a the table of contents is empty (no object file members in the library define global symbols)
npm ERR! /bin/sh: -c: line 0: syntax error near unexpected token `('
npm ERR! /bin/sh: -c: line 0: `c++ -o Release/obj.target/sharp/src/common.o ../src/common.cc '-DNODE_GYP_MODULE_NAME=sharp' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DNAPI_VERSION=3' '-DBUILDING_NODE_EXTENSION' -I/Users/raguay/Library/Caches/node-gyp/16.4.1/include/node -I/Users/raguay/Library/Caches/node-gyp/16.4.1/src -I/Users/raguay/Library/Caches/node-gyp/16.4.1/deps/openssl/config -I/Users/raguay/Library/Caches/node-gyp/16.4.1/deps/openssl/openssl/include -I/Users/raguay/Library/Caches/node-gyp/16.4.1/deps/uv/include -I/Users/raguay/Library/Caches/node-gyp/16.4.1/deps/zlib -I/Users/raguay/Library/Caches/node-gyp/16.4.1/deps/v8/include -I../../node-addon-api -I/Users/raguay/Dropbox (Personal)/Richard/rag/JavaScript/svelte/EmailIt/node_modules/sharp/vendor/8.10.6/include -I/Users/raguay/Dropbox (Personal)/Richard/rag/JavaScript/svelte/EmailIt/node_modules/sharp/vendor/8.10.6/include/glib-2.0 -I/Users/raguay/Dropbox (Personal)/Richard/rag/JavaScript/svelte/EmailIt/node_modules/sharp/vendor/8.10.6/lib/glib-2.0/include  -O3 -gdwarf-2 -mmacosx-version-min=10.9 -arch arm64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++11 -stdlib=libc++ -fexceptions -Wall -O3 -MMD -MF ./Release/.deps/Release/obj.target/sharp/src/common.o.d.raw   -c'
npm ERR! make: *** [Release/obj.target/sharp/src/common.o] Error 2
npm ERR! gyp ERR! build error
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:394:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
npm ERR! gyp ERR! System Darwin 20.5.0
npm ERR! gyp ERR! command "/opt/homebrew/Cellar/node/16.4.1/bin/node" "/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /Users/raguay/Dropbox (Personal)/Richard/rag/JavaScript/svelte/EmailIt/node_modules/sharp
npm ERR! gyp ERR! node -v v16.4.1
npm ERR! gyp ERR! node-gyp -v v7.1.2
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/raguay/.npm/_logs/2021-07-03T02_51_25_939Z-debug.log

**Comment by leimantas** [comment]
Try YARN. Everything works for me with yarn.

**Comment by m2lab** [comment]
It works fine on a Mac Mini M1 with Big Sur 11.4.
Your problem is probably related to the architecture: you can't compile arm64 apps yet (afaik). The rust build target is Intel. It still builds much faster than on my Intel computers 😉.
Sorry can't help with details, just￼￼ wanted to reassure you that it works very well.

**Comment by nothingismagick** [comment]
Or use the `cargo cli` -> this problem seems to be with sharp, unfortunately. It is only required for making your icons.

```
% cargo install tauri-cli --version ^1.0.0-beta
% cargo tauri --version
```

With that you can run `cargo tauri dev|build|info`

**Comment by raguay** [comment]
Hi,

Thanks. I was able to get it running with the `cargo install tauri-cli
--version ^1.0.0-beta` command line. But, It wouldn't work in fish shell. I
had to load zsh to run the command.  The carrot is processed by fish as a
pipe. I wasn't sure if the cargo command line would take the version with
quotes, so I just ran it as zsh. I'm new to Rust and it's tools (other than
just building my favorite task runner: mask).

Richard
